curated_transformers/models/attention.py

```diff
@@ -115,6 +116,20 @@ def filter_batch_items(self, mask: Tensor) -> "KeyValueCache":
         return KeyValueCache(key=self.key[mask], value=self.value[mask])
 
 
+class QkvMode(IntEnum):
+    """Modes of handling the query, key and value projections
+    in the self-attention layer.
+
+    ``SEPARATE`` - Use separate projections for query, key and value.
+    ``MERGED_SPLIT_BEFORE`` - Use a merged projection for query, key and value, and split heads before chunking.
+    ``MERGED_SPLIT_AFTER`` - Use a merged projection for query, key and value, and split heads after chunking.
```

Comment:
I wonder how this is rendered once we expose this in the docs? Aren't these lines rendered as a single paragraph, maybe it's worth prefixing it by `- ` to get a bulleted list. Or perhaps the `#: ` here on the fields, since they are not constructor arguments here?

Reply:
Oh yeah, it does appear as a single paragraph. Moved the string to the variants.